### PR TITLE
Rename ParticleHandler::locally_relevant_ids() to locally_owned_particles()

### LINKED ISSUE #10263: ParticleHandler documentation issue
@luca-heltai @blaisb -- FYI

The `ParticleHandler` has a member function like this:
```
    /**
     * Extract an IndexSet with global dimensions equal to
     * get_next_free_particle_index(), containing the locally owned
     * particle indices.
     *
     * This function can be used to construct distributed vectors and matrices
     * to manipulate particles using linear algebra operations.
     *
     * Notice that it is the user's responsability to guarantee that particle
     * indices are unique, and no check is performed to verify that this is the
     * case, nor that the union of all IndexSet objects on each mpi process is
     * complete.
     *
     * @return An IndexSet of size get_next_free_particle_index(), containing
     * n_locally_owned_particle() indices.
     *
     * @author Luca Heltai, Bruno Blais, 2019.
     */
    IndexSet
    locally_relevant_ids() const;
```
What's a bit confusing is that the function documentation talks about "locally owned particles" whereas the function name mentions "locally relevant". We should clarify.

### PR DESCRIPTION
**Pull request by peterrum**
closes #10263

I took the liberty to directly deprecate the function, since it is a rather new function.

FYI @bangerth @gassmoeller @luca-heltai @blaisb

### CODE REVIEW

File: include/deal.II/particles/particle_handler.h [hunk 1]
```diff
@@ -668,7 +689,7 @@ namespace Particles
      * n_locally_owned_particle() indices.
      */
     IndexSet
-    locally_relevant_ids() const;
+    locally_owned_particles() const;
```
Comment (bangerth) [inline include/deal.II/particles/particle_handler.h:692]: I think that the function should be name to make it clear that it doesn't return *particles* but *particle ids*:
```suggestion
    locally_owned_particle_ids() const;
```

**Review by bangerth — CHANGES_REQUESTED** [review]
I think the function should have a different name, but other than that this is ok.

**Comment by peterrum** [comment]
@bangerth Done!

**Review by marcfehling — APPROVED** [review]

**Review by bangerth — APPROVED** [review]

File: include/deal.II/particles/particle_handler.h [hunk 2]
```diff
@@ -641,7 +641,7 @@ namespace Particles
      * triangulation.
      */
     types::particle_index
-    n_locally_owned_particles() const;
+    n_locally_owned_particle_ids() const;
```
Comment (gassmoeller) [inline include/deal.II/particles/particle_handler.h:644]: I know I am late to this PR, sorry for that. Why was this function name changed as well? This is not really part of the index discussion, and was a function that was in here for some years. This renaming also makes it inconsistent with the `n_global_particles()` function above. Unless there is a specific reason for this change, I would prefer if we could revert this particular change (only this function name). (Also see the problems reported in https://github.com/geodynamics/aspect/issues/4023#issuecomment-832036067).

Comment (peterrum) [reply include/deal.II/particles/particle_handler.h:644]: This was an accident. Sorry. PR is coming!

Comment (gassmoeller) [reply include/deal.II/particles/particle_handler.h:644]: No harm done, thanks for the quick reply!

Comment (bangerth) [reply include/deal.II/particles/particle_handler.h:644]: Oh, and I totally missed it too :-(

Comment (blaisb) [reply include/deal.II/particles/particle_handler.h:644]: Thanks @gassmoeller . Good catch :).

**Pull request merged**